examples/year_playlist.py

```diff
@@ -0,0 +1,145 @@
+#!/usr/bin/env python3
+
+"""Script for retrieving songs from a user's 'Your Top Songs 2017'
+playlist.
+
+Note: Requests to Spotify API do not currently include relevant fields arg,
+which should eventually be added.
+"""
+
+from collections import namedtuple
+import sys
+import argparse
+import spotipy
+from spotipy import oauth2
+
+# Client ID and Client Secret for zachthehammer's spotipy-test API key
+_CLIENT_ID     = '5c243ed8edc9475a90bda2cc4a9a828c'
+_CLIENT_SECRET = 'b414caddc10143eea70db6edc1f4fb47'
+
+# Your Top Songs 2017 is actually owned by spotify user.
+_SPOTIFY_USER_ID    = 'spotify'
+_YEAR_PLAYLIST_NAME = 'Your Top Songs 2017'
+
+# Format strings
+_SCRIPT_DESCRIPTION_FMT = 'Get the list of tracks from a user\'s "{}" playlist.'
+_NO_PLAYLISTS_FOUND_FMT = 'No playlists found for user {}.'
+_PLAYLIST_NOT_FOUND_FMT = 'User "{user}"" does not follow their "{playlist}" playlist.'
+_USER_PLAYLIST_FMT      = '{user}\'s "{playlist}" playlist!'
+_TRACK_FMT              = '{artist} // {name} [{album}]'
+
+SpotifyTrack = namedtuple('SpotifyTrack', ['name', 'artist', 'album'])
+"""Namedtuple for spotify tracks."""
+
+def _parse_args():
+    """Helper function to create the command-line argument parser for
+    year_playlist.  Return the parsed arguments as a Namespace object if
+    successful. Exits the program if unsuccessful or if the help
+    message is printed.
+    """
+    description = (_SCRIPT_DESCRIPTION_FMT.format(_YEAR_PLAYLIST_NAME))
+    parser = argparse.ArgumentParser(description=description)
+
+    parser.add_argument('username',
+                        help='User\'s username.')
+
+    return parser.parse_args()
+
+def _get_user_playlists(spotify_client, username):
+    """Get the list of playlists of a user as raw dictionaries. """
+    playlists = spotify_client.user_playlists(user=username)
+    return playlists and playlists['items']
+
+def _get_playlist_by_name(playlists, name):
+    """Get a playlist by its name from a list of raw playlist dictionaries."""
+    found_playlist = None
+    for playlist in playlists:
+        if playlist['name'] == name:
+            found_playlist = playlist
+            break
+    return found_playlist
+
+def _extract_track_data(raw_track):
+    """Extract relevant track data from a track's raw dictionary representation
+    and return it is a SpotifyTrack object."""
+    return SpotifyTrack(name=raw_track['track']['name'],
+                        artist=raw_track['track']['artists'][0]['name'],
+                        album=raw_track['track']['album']['name'])
+
+def _get_tracks_of_playlist(spotify_client, playlist_id, owner_id):
```

Comment:
I try to reserve `get` for accessor type functions: no IO, little calculations, but that might just be my Java background.

How about `fetch` or `read`?